src/XMakeBuildEngine/UnitTests/Evaluation/ImportFromMSBuildExtensionsPath_Tests.cs

```diff
@@ -167,6 +168,47 @@ public void ImportFromExtensionsPathWithWildCard()
         }
 
         [Fact]
+        public void ImportFromExtensionsPathWithWildCardAndSelfImport()
```

Comment:
```
                 if (result == LoadImportsResult.FoundFilesToImportButIgnored)
                  {
                      // Circular, Self import cases are usually ignored
                      // Since we have a semi-success here, we stop looking at
                      // other paths
-                      return projects;
```

The old code would try the search path `extensions2`, hit the circular import and the above code path would cause it to *stop* the search and return at that point. But the new code ...

```
 +                    // If we don't have a wildcard and we had a match, we're done.
 +                    if (!containsWildcards)
 +                    {
 +                        return projects;
 +                    }
 +
 +                    allProjects.AddRange(projects);
``` 

... does not stop, in case of a wildcard. For that, the test here has an file available in `extensions3`, which gets imported only for the new code and not with the old one. Using `extnTargetsFileContent` without the Import should be fine for `extn3.proj`. Maybe we can add a comment about this in the test, to make it explicit.